DatadogInternal/Sources/NetworkInstrumentation/OpenTracing/HTTPHeadersReader.swift

```diff
@@ -6,26 +6,26 @@
 
 import Foundation
 
-internal class HTTPHeadersReader: OTHTTPHeadersReader, TracePropagationHeadersExtractor {
+public class HTTPHeadersReader: TracePropagationHeadersReader {
     private let httpHeaderFields: [String: String]
 
-    init(httpHeaderFields: [String: String]) {
+    public init(httpHeaderFields: [String: String]) {
         self.httpHeaderFields = httpHeaderFields
     }
 
-    func extract() -> OTSpanContext? {
+    public func read() -> (traceID: TraceID, spanID: SpanID, parentSpanID: SpanID?)? {
```

Comment:
I like that!

Reply:
So the `OTSpanContext` will be part of the `DatadogTrace` module only.
The `HTTPHeadersReader` is shared in `DatadogInternal` to enable trace propagation in `DatadogTrace` or `DatadogRUM`, we only need the ids here, without the entire span logic 👍